spec/controllers/api/v2/providers_controller_spec.rb

```diff
@@ -31,10 +31,22 @@
       let(:course) { create(:course) }
       let!(:other_provider) { course.provider }
 
-      it "returns the specified provider" do
+      it "returns unassociated providers" do
         get :suggest_any, params: { query: other_provider.provider_code }
         expect(JSON.parse(response.body).dig("data").map { |p| p["id"] }).to eql([other_provider.id.to_s])
       end
     end
+
+    context "with a filter" do
+      let(:user) { provider.users.first }
+      let!(:provider) { create(:provider) }
+      let!(:in_scope_provider) { create(:provider, :accredited_body, provider_name: "in provider") }
+      let!(:out_scope_provider) { create(:provider, provider_name: "out provider") }
+
+      it "applies filter to results" do
+        get :suggest_any, params: { query: "provider", filter: { accredited_body: "true" } }
```

Comment:
where is the false test?

Reply:
I can add a false test. It is kind of implied here https://github.com/DFE-Digital/teacher-training-api/pull/1425/files#diff-deffb905a7d19149f046988a181c4ae1R34 when the value is not provided